modules/popup/react/lib/Popper.tsx

```diff
@@ -33,29 +41,63 @@ export interface PopperProps extends React.HTMLAttributes<HTMLDivElement> {
    */
   popperOptions?: Partial<PopperOptions>;
   /**
-   * If true, attach the Popper to the `containerElement`. If false, render the Popper within the DOM hierarchy of its parent.
+   * If true, attach the Popper to the `containerElement`. If false, render the Popper within the
+   * DOM hierarchy of its parent.
    * @default true
    */
   portal?: boolean;
+  type?: 'ephemeral' | 'persistent';
 }
 
 export const Popper = React.forwardRef<HTMLDivElement, PopperProps>(
   (
     {
-      placement: popperPlacement = 'bottom',
-      popperOptions = {},
+      placement,
+      popperOptions,
       portal = true,
       open = true,
       anchorElement,
       children,
       containerElement,
+      type,
       ...elemProps
     }: PopperProps,
-    forwardRef
+    ref
+  ) => {
+    if (!open) {
+      return null;
+    }
+
+    const contents = (
+      <OpenPopper
+        {...{ref, type, anchorElement, popperOptions, placement, children, ...elemProps}}
+      />
+    );
+
+    if (!portal) {
+      return contents;
+    }
+
+    return ReactDOM.createPortal(contents, containerElement || document.body);
+  }
+);
+
+const OpenPopper = React.forwardRef<HTMLDivElement, PopperProps>(
```

Comment:
Can you add some documentation on why you had to break these out? It's a bit confusing at first glance